### DISCUSSION TITLE
How can I implement a non React reaction (side effect) after an atom or selector change?

### DISCUSSION THREAD

**Discussion by zisiszikos** [facebookexperimental/Recoil#1202] [Q&A]
Assuming I have an **Atom** represanting a list like:
```js
const myList = atom({
  key: 'myList',
  default: [],
});
```
and there are a few places in my code that this array changes (add or remove item).
How can I write some code like a **Reaction** (side effect) that runs every time this atom's array change? And I mean without using React at all (useEffect, useCallback, useRecoilCallback).

For example, MobX has `reaction`s, redux has something like `thunk`s or `saga`s.

**Comment by BenjaBobs** [comment]
You can use [Atom Effects](https://recoiljs.org/docs/guides/atom-effects) for this.

**Reply by zisiszikos** [reply]
Thanks for your answer.

I tried to make an example app that uses Atom Effects and I couldn't make it work. The problem is that I cannot get or set the value of another Atom from inside an Atom Effect function.

Consider the case where the selected item of a list must conditionally change whenever an item is removed from the list.

Here is an example code: [example](https://codesandbox.io/s/react-recoil-side-effect-cuxit?file=/src/App.js)
